lib/models/ObjectMD.js

```diff
@@ -753,6 +753,19 @@ class ObjectMD {
     }
 
     /**
+     * Get dataStoreVersionId
+     *
+     * @return {string} external backend version id for data
+     */
+    getDataStoreVersionId() {
+        if (!this._data.location || !Array.isArray(this._data.location)
```

Comment:
You can call `getLocation()` which always returns an array (possibly empty), and get rid of the first two checks in this test.

Reply:
When metadata is loaded from the blob, `_convertToLatestModel()` ensures that `location` is or becomes an array or `null`.

Reply:
updated -- you're right, but don't I still one of the two checks in case it is null? Otherwise we would get a `Cannot read property '0' of null` error

Reply:
nvm I see what you are saying and will update again